### PR TITLE
[`Doctest`] Fix pix2struct doctest

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Link to failing job: https://github.com/huggingface/transformers/actions/runs/4867713745/jobs/8680544136

This PR fixes the current failing doctest for pix2struct. https://github.com/huggingface/transformers/pull/23051 fixed the issues related with pix2struct and training by changing the way attention masks are computed. Naturally this has changed the value of the expected loss function in the docstring

cc @ydshieh

### CODE REVIEW

**Review by ydshieh — APPROVED** [review]
Thank you. Good for me, and we can merge. But as discussed offline, we have to discuss the usage of `is_decoder`.
